sda-commons-client-jersey-wiremock-testing/src/main/java/org/sdase/commons/client/jersey/wiremock/testing/WireMockExtension.java

```diff
@@ -0,0 +1,82 @@
+package org.sdase.commons.client.jersey.wiremock.testing;
+
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
+
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.client.VerificationException;
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.core.Options;
+import com.github.tomakehurst.wiremock.verification.LoggedRequest;
+import com.github.tomakehurst.wiremock.verification.NearMiss;
+import java.util.List;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Junit 5 replacement for {@link com.github.tomakehurst.wiremock.junit.WireMockRule} */
+public class WireMockExtension extends WireMockServer
+    implements BeforeEachCallback, AfterEachCallback {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(WireMockExtension.class);
+
+  private final boolean failOnUnmatchedRequests;
+
+  public WireMockExtension(Options options) {
+    this(options, true);
+  }
+
+  public WireMockExtension(Options options, boolean failOnUnmatchedRequests) {
```

Comment:
Can you provide the simpler constructors in combination with `failOnUnmatchedRequests` as well?

Reply:
Actually I would prefer a method `assertAllRequestsMatched()`. Then each test case can add an explizit assertion and it can be used with a class extension as well.